### PR TITLE
[bugfix]ZIOS-9051-Opening-conversation-from-Contacts-UI-lead-to-an-empty-screen(Xcode9)

### PR DESCRIPTION
**Pull request by billypchan**
fix ConversationListViewController alpha = 0 issue.
check self.presentedViewController != nil before dismiss.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController.m [hunk 1]
```diff
@@ -327,7 +327,16 @@ - (void)setState:(ConversationListState)state animated:(BOOL)animated completion
 
     switch (state) {
         case ConversationListStateConversationList: {
-            [self.presentedViewController dismissViewControllerAnimated:YES completion:completion];
+            self.view.alpha = 1;
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController.m]: Shouldn't this move into the `else` statement?

**Comment by typfel** [comment]
You're changes here are good and make the this code more robust but as you said in the comments in https://wearezeta.atlassian.net/browse/ZIOS-9051

The root cause for this is that the dismiss is called twice. Shouldn't we then also remove the dismiss  in the `InviteContactsViewController` since it anyway is called by `selectConversation:focusOnView:animated:`

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController.m (hunk 1 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController.m]: Not sure, but I think it is logical to always make self.view.alpha = 1 when someone want to change the state to ConversationListStateConversationList (of ConversationListViewController).

**Comment by billypchan** [comment]
@typfel Should we always dismiss the view from ZClientViewController?

File: Wire-iOS/Sources/UserInterface/InviteContacts/InviteContactsViewController.m [hunk 2]
```diff
@@ -57,21 +57,19 @@ - (BOOL)sharingContactsRequired
...
 {
     if (user.isConnected) {
-        [self dismissViewControllerAnimated:YES completion:^{
             [[ZClientViewController sharedZClientViewController] selectConversation:user.user.oneToOneConversation
                                                                         focusOnView:YES
                                                                            animated:YES];
-        }];
     } else if (user.user.isPendingApprovalBySelfUser && ! user.user.isIgnored) {
-        [self dismissViewControllerAnimated:YES completion:^{
+//        [self dismissViewControllerAnimated:YES completion:^{
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/InviteContacts/InviteContactsViewController.m]: commented out code

**Pull request merged**